src/app/inventory/item-types.ts

```diff
@@ -282,9 +282,10 @@ export interface DimCatalyst {
   complete: boolean;
   /** Whether the player has unlocked/discovered the catalyst */
   unlocked: boolean;
+  /** Progress the player has made on unlocking the catalyst  */
+  objectives?: DestinyObjectiveProgress[];
 
   // TODO: on item popup, remove catalyst socket if item.catalystInfo is undefined
```

Comment:
this todo can be removed as well. the catalyst socket does not show on all items anymore